mimirtool: retain old default of blocks_storage.backend

dimitarvdimitrov opened the pull request:
#### What this PR does

When passing `--update-defaults` to mimirtool config convert and the
value of blocks_storage.backend is using the old default of "s3", the
value in the resulting config is updated to "filesystem." It is better
if the value is not updated.


#### Which issue(s) this PR fixes or relates to

Fixes #1622 

#### Checklist

- [x] Tests updated
- [na] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

pstibrany reviewed (commented):
What about other storages? (Alertmanager, ruler) I would expect the update path to be the same for all of them (if alertmanager / ruler were used in the first place).

pstibrany commented:
I think this now deserves a CHANGELOG entry too.

dimitarvdimitrov commented:
nice call @pstibrany. I've added `(ruler|alertmanager)_storage.backend` to the list of retained old defaults in https://github.com/grafana/mimir/pull/1626/commits/740878d0722073cc56fb5d8e1e57924483a27399 and added a changelog entry in https://github.com/grafana/mimir/pull/1626/commits/fbbc35514b14ca5fdc87c5d81dd4cacd7145144c 

I also added some docs to mention the special behaviour for these defaults in https://github.com/grafana/mimir/pull/1626/commits/a075db07946d5d1a2864fc36bfa5a237da3a6531. Also added `type/docs` label to get some advice from Docs squad on that paragraph.

pstibrany commented on docs/sources/operators-guide/tools/mimirtool.md:
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
+- `ruler_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
+- `alertmanager_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
+- `server.http_listen_port` - Cortex default is `80`, Mimir default is `8080`
+- (GEM only) `auth.type` - Cortex default is `trust`, Mimir default is `enterprise`
```
Nit: It's GEM default, not Mimir.

pstibrany reviewed (approved):
Looks good to me. There is no good / easy way checking if alertmanager and ruler were used before (in Cortex) :(

pracucci reviewed (approved):
I'm not much familiar with the config conversion tool, but to my understanding the changes make sense.

dimitarvdimitrov commented:
thank you for reviewing. For visibility - I'm waiting for the docs squad to get to this PR so they can review the added docs to mimirtool. 

There is other work related to this that I'd like to do - some filesystem paths (like `/data/compactor` changed default values, and we can employ a similar approach as in this PR to provide a smoother migration.

I want these to be included in the 2.1 release, so if I don't get a docs review in the next week. We can extract the docs in a seperate PR and move on with the code changes

osg-grafana commented on docs/sources/operators-guide/tools/mimirtool.md:
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
 | `-v`, `--verbose`    | If you set this flag, the CLI flags and YAML paths from the old configuration that do not exist in the new configuration are printed to `stderr`. This flag also prints default values that have changed between the old and the new configuration. |
 | `--gem`              | If you set this flag, the tool will convert from Grafana Metrics Enterprise (GEM) v1.7.x to v2.0.0.                                                                                                                                                 |
 
+##### A note on default values
```
```suggestion
##### Changes to default values
```

osg-grafana commented on docs/sources/operators-guide/tools/mimirtool.md:
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
 
+##### A note on default values
+
+This command is designed to aid a migration from Cortex to Grafana Mimir. There are changes to the default values of
```
```suggestion
The `mimirtool` command helps you migrate from Cortex to Grafana Mimir. There are changes to the default values of
```
I assume we are talking about the `mimirtool` here.

osg-grafana commented on docs/sources/operators-guide/tools/mimirtool.md:
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
+##### A note on default values
+
+This command is designed to aid a migration from Cortex to Grafana Mimir. There are changes to the default values of
+some configuration parameters in Mimir v2.0.0 that we think you would not want to make as part of this migration. These
```
```suggestion
some configuration parameters in Mimir v2.0.0 that you might not want to use as part of this migration:
```

osg-grafana commented on docs/sources/operators-guide/tools/mimirtool.md:
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
+
+This command is designed to aid a migration from Cortex to Grafana Mimir. There are changes to the default values of
+some configuration parameters in Mimir v2.0.0 that we think you would not want to make as part of this migration. These
+parameters are:
```
```suggestion
```

osg-grafana commented on docs/sources/operators-guide/tools/mimirtool.md:
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
+- `server.http_listen_port` - Cortex default is `80`, Mimir default is `8080`
+- (GEM only) `auth.type` - GEM 1.x default is `trust`, 2.x default is `enterprise`
+
+This command will output the Cortex default value even when the configuration parameter is not explicitly set in the input
+configuration. If the Cortex default value is explicitly set in the input configuration, and you have provided
+the `--update-defaults` flag, the command will not update the value to the Mimir default.
```
```suggestion
The `mimirtool` command outputs the Cortex default value even when the configuration parameter is not explicitly set in the input
configuration. If in the input configuration you explicitly set the Cortex default value, and you have provide
the `--update-defaults` flag, the `mimirtool` command will not update the value to the Mimir default.
```
Please vet for technical accuracy.

dimitarvdimitrov replied:
changed in cecbdd27675f13e9375b09289347cb4186623398, can you take another look?

osg-grafana reviewed (commented):
Unblocking with feedback, which I would like in part to be technically vetted for accuracy.

osg-grafana reviewed (approved):
See previous comment. :)
